### PR TITLE
[ISSUE #4225]  Replace http client in HttpHealthCheckProcessor

### PR DESCRIPTION
**Pull request by Maijh97**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

fix: #4225 

## Brief changelog

Remove the `com.ning Async HTTP client` and replace it with a unified HTTP client.

## Verifying this change

1. Register a persistent service
2. Use HTTP health checks
3. Health check normal

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: common/src/main/java/com/alibaba/nacos/common/http/AbstractHttpClientFactory.java [hunk 1]
```diff
@@ -70,20 +71,28 @@ public void onChanged(String filePath) {
     public NacosAsyncRestTemplate createNacosAsyncRestTemplate() {
         final HttpClientConfig originalRequestConfig = buildHttpClientConfig();
         final RequestConfig requestConfig = getRequestConfig();
+        final IOReactorConfig ioReactorConfig = getIoReactorConfig();
         return new NacosAsyncRestTemplate(assignLogger(), new DefaultAsyncHttpClientRequest(
                 HttpAsyncClients.custom()
                         .addInterceptorLast(new RequestContent(true))
+                        .setDefaultIOReactorConfig(ioReactorConfig)
```
Comment (KomachiSion) [inline common/src/main/java/com/alibaba/nacos/common/http/AbstractHttpClientFactory.java:78]: use getIoReactorConfig directly, no need to use attribute.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Review by KomachiSion — DISMISSED** [review]

File: naming/src/main/java/com/alibaba/nacos/naming/misc/HttpClientManager.java [hunk 2]
```diff
@@ -146,4 +163,27 @@ protected Logger assignLogger() {
             return SRV_LOG;
         }
     }
+    
+    private static class ProcessorHttpClientFactory extends AbstractHttpClientFactory {
```
Comment (chuntaojun) [inline naming/src/main/java/com/alibaba/nacos/naming/misc/HttpClientManager.java:167]: 这里的参数应当后面可以对外可配置化的，而不是写死的

**Review by sanxun0325 — DISMISSED** [review]

File: naming/src/main/java/com/alibaba/nacos/naming/misc/HttpClientManager.java [hunk 3]
```diff
@@ -131,16 +147,58 @@ protected Logger assignLogger() {
...
+        
+        @Override
+        protected Logger assignLogger() {
+            return SRV_LOG;
         }
+    }
     
+    private static class ProcessorHttpClientFactory extends AbstractHttpClientFactory {
+        
+        private static final int CONNECTION_REQUEST_TIMEOUT = EnvUtil
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/misc/HttpClientManager.java:166]: Static final variable get from properties? Static final means that it's a constants. Should not be configed.

File: sys/src/main/java/com/alibaba/nacos/sys/env/Constants.java [hunk 4]
```diff
@@ -63,4 +63,33 @@
     String NACOS_SERVER_HEADER = "Nacos-Server";
     
     String REQUEST_PATH_SEPARATOR = "-->";
+    
+    /**
+     * ============ naming persistent health check http env.
+     */
+    String NAMING_PERSISTENT = "nacos.naming.persistent";
```
Comment (KomachiSion) [inline sys/src/main/java/com/alibaba/nacos/sys/env/Constants.java:70]: Naming Constants should not be in parent module.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**